packages/@aws-cdk/aws-apigatewayv2/lib/websocket/api.ts

```diff
@@ -0,0 +1,156 @@
+import * as crypto from 'crypto';
+import { IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnApi } from '../apigatewayv2.generated';
+import { WebSocketRouteIntegrationConfig, WebSocketIntegration } from './integration';
+import { WebSocketRoute, WebSocketRouteOptions } from './route';
+import { WebSocketStage } from './stage';
+
+/**
+ * Represents a WebSocket API
+ */
+export interface IWebSocketApi extends IResource {
+  /**
+   * The identifier of this API Gateway WebSocket API.
+   * @attribute
+   */
+  readonly webSocketApiId: string;
+
+  /**
+   * The default endpoint for an API
+   * @attribute
+   */
+  readonly apiEndpoint: string;
+
+  /**
+   * Add a websocket integration
+   * @internal
+   */
+  _addIntegration(config: WebSocketRouteIntegrationConfig): WebSocketIntegration
+}
+
+/**
+ * Props for WebSocket API
+ */
+export interface WebSocketApiProps {
+  /**
+   * Name for the WebSocket API resoruce
+   * @default - id of the WebSocketApi construct.
+   */
+  readonly apiName?: string;
+
+  /**
+   * The description of the API.
+   * @default - none
+   */
+  readonly description?: string;
+
+  /**
+   * The route selection expression for the API
+   * @default '$request.body.action'
+   */
+  readonly routeSelectionExpression?: string;
+
+  /**
+   * The name of the default stage with deployment
+   * @default - none
+   */
+  readonly defaultStageName?: string;
+}
+
+/**
+ * Create a new API Gateway WebSocket API endpoint.
+ * @resource AWS::ApiGatewayV2::Api
+ */
+export class WebSocketApi extends Resource implements IWebSocketApi {
+  public readonly webSocketApiId: string;
+  public readonly apiEndpoint: string;
+
+  /**
+   * A human friendly name for this WebSocket API. Note that this is different from `webSocketApiId`.
+   */
+  public readonly webSocketApiName?: string;
+
+  /**
+   * default stage of the api resource
+   */
+  public readonly defaultStage: WebSocketStage | undefined;
+
+  private integrations: Record<string, WebSocketIntegration> = {};
+
+  constructor(scope: Construct, id: string, props?: WebSocketApiProps) {
+    super(scope, id);
+
+    this.webSocketApiName = props?.apiName ?? id;
+
+    const resource = new CfnApi(this, 'Resource', {
+      name: this.webSocketApiName,
+      protocolType: 'WEBSOCKET',
+      description: props?.description,
+      routeSelectionExpression: props?.routeSelectionExpression ?? '$request.body.action',
+    });
+    this.webSocketApiId = resource.ref;
+    this.apiEndpoint = resource.attrApiEndpoint;
+
+    if (props?.defaultStageName) {
+      this.defaultStage = new WebSocketStage(this, 'DefaultStage', {
+        webSocketApi: this,
+        stageName: props.defaultStageName,
+        autoDeploy: true,
+      });
+    }
+  }
+
+  /**
+   * @internal
+   */
+  public _addIntegration(config: WebSocketRouteIntegrationConfig): WebSocketIntegration {
```

Comment:
Wondering if we could move this into the common area for sharing between http and websocket apis. We already have a `common/` folder that is meant to hold some of these.

Maybe we introduce an `ApiBase` private class that is shared between the two? You could also (if needed) introduce an `IApi` interface that is the parent of `IHttpApi` and `IWebsocketApi` that has the `_addIntegration()` method.